rewrite/rewrite/python/format/spaces_visitor.py

```diff
@@ -0,0 +1,236 @@
+        """
+        a: ChainedAssignment = cast(ChainedAssignment, super().visit_chained_assignment(chained_assignment, p))
+        a = a.padding.with_variables(
+            [v.with_after(update_space(v.after, self._style.around_operators.assignment)) for v in a.padding.variables])
+
+        a = a.padding.with_variables(
+            [v.with_element(
+                self.space_before(v.element, self._style.around_operators.assignment if idx >= 1 else False)) for idx, v
+                in enumerate(a.padding.variables)])
+
+        return a.with_assignment(self.space_before(a.assignment, self._style.around_operators.assignment))
+
+    def visit_member_reference(self, member_reference: MemberReference, p: P) -> J:
+        m: MemberReference = cast(MemberReference, super().visit_member_reference(member_reference, p))
+
+        if m.padding.type_parameters is not None:
+            # TODO: Handle type parameters, relevant for constructors in Python.
+            raise NotImplementedError("Type parameters are not supported yet")
+
+        return m
+
+    def visit_binary(self, binary: j.Binary, p: P) -> J:
+        b: j.Binary = cast(j.Binary, super().visit_binary(binary, p))
+        op: j.Binary.Type = b.operator
+
+        # Additive operators, _self._style.around_operators.addition is True by default
+        if op in [j.Binary.Type.Addition, j.Binary.Type.Subtraction]:
+            b = self._apply_binary_space_around(b, self._style.around_operators.additive)
+        elif op in [j.Binary.Type.Multiplication, j.Binary.Type.Division, j.Binary.Type.Modulo]:
+            b = self._apply_binary_space_around(b, self._style.around_operators.multiplicative)
+        elif op in [j.Binary.Type.LessThan, j.Binary.Type.GreaterThan, j.Binary.Type.LessThanOrEqual,
+                    j.Binary.Type.GreaterThanOrEqual, j.Binary.Type.Equal, j.Binary.Type.NotEqual]:
+            b = self._apply_binary_space_around(b, self._style.around_operators.relational)
+        elif op in [j.Binary.Type.BitAnd, j.Binary.Type.BitOr, j.Binary.Type.BitXor]:
+            b = self._apply_binary_space_around(b, self._style.around_operators.bitwise)
+        elif op in [j.Binary.Type.LeftShift, j.Binary.Type.RightShift, j.Binary.Type.UnsignedRightShift]:
+            b = self._apply_binary_space_around(b, self._style.around_operators.shift)
+        elif op in [j.Binary.Type.Or, j.Binary.Type.And]:
+            b = self._apply_binary_space_around(b, True)
+        else:
+            raise NotImplementedError(f"Operation {op} is not supported yet")
+
+        return b
+
+    def visit_python_binary(self, binary: Binary, p: P) -> J:
+        b: Binary = cast(Binary, super().visit_python_binary(binary, p))
+        op: Binary.Type = b.operator
+
+        if op == Binary.Type.In or op == Binary.Type.Is or op == Binary.Type.IsNot or op == Binary.Type.NotIn:
+            # TODO: Not sure what style options to use for these operators
```

Comment:
Your current implementation is correct.
```suggestion
            # A space is required because the operators are identifiers
```